### DISCUSSION TITLE
verdaccio next (WIP)

### DISCUSSION THREAD

**Discussion by juanpicado** [verdaccio/verdaccio#2970] [Show and tell]
> ℹ️  The legacy verdaccio 6 naming would became `next`, thus the current v5 can easily assume major changes without major issue. [Many modules from the next are already being used on v5](https://github.com/verdaccio/verdaccio/blob/5.x/package.json#L22), thus 5.x is not entirely 5.x only anymore.

List of version next changes, once all this tickets are done v6 could be released, more improvements are listed on the [roadmap](https://github.com/verdaccio/verdaccio/discussions/1690) but are welcome on this release.

### Install

```bash
npm install --global verdaccio@6-next
```

or

```bash
docker pull verdaccio/verdaccio:nightly-master
```

* [All merged PR for v.next](https://github.com/verdaccio/verdaccio/milestone/25?closed=1)
* [Migration notes](https://github.com/verdaccio/verdaccio/blob/master/docs/migration-v5-to-v6.md)

### 💥  Breaking changes 
- Node 16 minimum #2833 #2473
- Experiments 
   -  [ ] Uses Fastify (experimental) server https://github.com/verdaccio/verdaccio/issues/2623   [see all tickets](https://github.com/verdaccio/verdaccio/issues?q=+label%3A%22fastify+migration%22+) 🚧 
- Search API v1 #2431 #1839 
- Web search endpoint new structure https://github.com/verdaccio/verdaccio/pull/3057 
- New plugins API, `async/await` instead callback #3100 
### 🐛 Bugfix

- #2967
- [Documentation improvements](https://github.com/verdaccio/verdaccio/issues?q=label%3A%22fastify+migration%22+milestone%3A%22v6+release%22)
- #2652 
- #2650 
- #2605
- #2446
- #2335
- https://github.com/verdaccio/verdaccio/pull/2193
- https://github.com/verdaccio/verdaccio/issues/1778
- [x] https://github.com/verdaccio/verdaccio/issues/2616
- [x] Removed `request` #1807 with #3100 @juanpicado  #3100 
- [x] `async` #1225  with #3100 @juanpicado  #3100 

### 🌟 Features
- Web
   - #2936 
   - #2913 
   - UI refactors #2612 #2447 #2295
   - #2228
   - https://github.com/verdaccio/verdaccio/pull/2209
   - https://github.com/verdaccio/verdaccio/pull/2226
   - [x] Use the new search v1 endpoint via UI (to search any package, private and public from the UI) https://github.com/verdaccio/verdaccio/pull/3057
- Server
   - Improve warning with codes #2821 #2855
   - #2835
   - https://github.com/verdaccio/verdaccio/pull/2199
   - https://github.com/verdaccio/verdaccio/pull/2159
   - https://github.com/verdaccio/verdaccio/pull/2072
   - https://github.com/verdaccio/verdaccio/pull/2050
   - https://github.com/verdaccio/verdaccio/pull/1999
   - [x] https://github.com/verdaccio/verdaccio/issues/1807
   - [x] https://github.com/verdaccio/verdaccio/issues/1225
   - [x] Introduce AbortSignals for request instead built-in solution #3100
   - [x] Migrate pino 8.x (currently using v6) https://github.com/verdaccio/verdaccio/pull/3308
- 🐳 🚢  Docker
  - #2689
  - [ ] https://github.com/verdaccio/verdaccio/issues/2457
  - [x] https://github.com/verdaccio/verdaccio/issues/2738
  - [ ] How to extend Docker images with `pnpm`, missign documentation and examples 
### ⌛ Features/Fixes pending to be moved to 6.x

- [x] [Some bufixes that were added to 5.x need to be moved to master, if this list is emtpy we are done](https://github.com/verdaccio/verdaccio/pulls?q=is%3Apr+label%3A%22pending+6.x%22+is%3Aclosed).
  -  [x] Part 1 https://github.com/verdaccio/verdaccio/pull/3461 

### 🏗️ Improvements for development o this project
- #2607
- https://github.com/verdaccio/verdaccio/pull/2515
- Crowdin improvements #2466 #2440 
- [More small fixes and refactors](https://github.com/verdaccio/verdaccio/milestone/25?closed=1)

**Comment by joergrech** [comment]
Hi, I would like to make some changes to verdaccio and am now wondering which version I should build on. Can you roughly say when v6 will be published? Is it in the range of weeks, months or more like 2023, etc?

**Reply by juanpicado** [reply]
Depends what you are looking for and the level of changes, I'd prefer keep v5 with bug-fixing and minor changes as possible. v6 still far away to be released (keep v5 freeze as possible helps as well), probably late this year but I have no a soon date at this point.

**Reply by juanpicado** [reply]
One more thing, just in case, the v5 and the v6 uses the [same UI code](https://github.com/verdaccio/verdaccio/blob/5.x/package.json#L25) (https://github.com/verdaccio/verdaccio/tree/master/packages/plugins/ui-theme) which helps a lot of fix bugs or releases early features only for the User Interface. 

Before it was hosted here https://github.com/verdaccio/ui but it was really hard to keep them on sync.
